internal/cmdsearch/putpattern_test.go

```diff
@@ -91,6 +91,74 @@ metadata:
   name: dev/my-project/nginx # kpt-set: ${env}/${project}/${name}
 spec:
   replicas: 3
+ `,
+	},
+	{
+		name: "put pattern by regex",
+		args: []string{"--by-value-regex", "my-project-*", "--put-pattern", "${project}-*"},
+		input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-project-deployment
+  namespace: my-project-namespace
+spec:
+  replicas: 3
+ `,
+		out: `${baseDir}/${filePath}
+fieldPath: metadata.name
+value: my-project-deployment # kpt-set: ${project}-deployment
+
+${baseDir}/${filePath}
+fieldPath: metadata.namespace
+value: my-project-namespace # kpt-set: ${project}-namespace
+
+Mutated 2 field(s)
+`,
+		expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-project-deployment # kpt-set: ${project}-deployment
+  namespace: my-project-namespace # kpt-set: ${project}-namespace
+spec:
+  replicas: 3
+ `,
+	},
+	{
+		name: "put pattern by regex, unable to derive values",
+		args: []string{"--by-value-regex", "something/*", "--put-pattern", "*/${image}:${tag}"},
+		input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: something/nginx:1.7.9
+ `,
+		out: `${baseDir}/${filePath}
+fieldPath: spec.template.spec.containers[0].image
+value: something/nginx:1.7.9 # kpt-set: */${image}:${tag}
```

Comment:
I'm thinking we should just make this an error. It seems unlikely that this is what users want. If we want to let them use `*` in the pattern we should provide a way to escape the `*`.

Reply:
Yes. I was also double-minded on that. I updated the PR to throw an error. We can iterate on the feedback.